src/services/formly.processor.ts

```diff
@@ -0,0 +1,23 @@
+
+import {FormlyFieldConfig} from "../components/formly.config";
+
+export interface FormlyConfigVisitor {
+  visit(config: FormlyFieldConfig);
+}
+
+export class FormlyConfigProcessor {
+
+  visitors: FormlyConfigVisitor[] = [new FormlyConfigValidator()];
+
+  process(fieldConfigs: FormlyFieldConfig[]): void {
+    fieldConfigs.forEach(field => {
+      this.visitors.forEach(visitor => {
+        visitor.visit(field);
+      });
+    });
+  }
+}
+
+class FormlyConfigValidator implements  FormlyConfigVisitor {
+  visit (field: FormlyFieldConfig) {}
+}
```

Comment:
Wanted to use Visitor pattern to create control group and also address #12 by having visitor also validate the config. But In final solution this was not used as found that individual template directive should hand control group creation. Left this implementation for future use of while addressing #12.